sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/JdbcTableDataObject.scala

```diff
@@ -147,15 +149,57 @@ case class JdbcTableDataObject(override val id: DataObjectId,
 
   override def getSparkDataFrame(partitionValues: Seq[PartitionValues] = Seq())(implicit context: ActionPipelineContext): DataFrame = {
     val queryOrTable = Map(table.query.map(q => ("query",q)).getOrElse("dbtable"->table.fullName))
-    val df = context.sparkSession.read.format("jdbc")
+    var df = context.sparkSession.read.format("jdbc")
       .options(options)
       .options(connection.getAuthModeSparkOptions)
       .options(queryOrTable)
       .load()
+    incrementalOutputState.foreach { case (lastColumn, lastWatermark)  =>
+      assert(incrementalOutputColumn.isDefined)
+      if (lastColumn != incrementalOutputColumn.get) logger.warn(s"($id) incrementalOutputState has different column as incrementalOutputColumn ($lastColumn != ${incrementalOutputColumn.get}")
```

Comment:
message confuses me. incrementalOutputState is lastColumn ?

Reply:
The incrementalOutputColumn configuration options is used to initialize State in the first run. SDLB keeps the column also as part of the state. In case the configuration option should change in the future, there is at least a warn message...